src/components/FilterBar/FilterBar.react.js

```diff
@@ -18,10 +18,6 @@ export class FilterBar extends React.Component {
       <div>
         <div>
           <div className="btn-group margin-bottom-sm">
-            <FilterList
-              disabledFilters={this.context.filterBarStore.getDisabled()}
-            />
```

Comment:
Yeah @SyntheticDave sorry, potentially I should put that architectural changes in PR description, because it's a huge changes. Definitely will do it before moving PR into Release stage.

So, before we had registered filters in a hash and it was represented in **_FilterBarStore_** like that:
```json
{
  "some_filter_key1": {
    "uid" : "some_filter_key1",
    "value" : "",
    "name" : "Some Filter Key 1",
    ...
  }, {
    "some_filter_key2" : {
    "uid" : "some_filter_key2",
    "value" : "",
    "name" : "Some Filter Key 2",
    ...
  }
}
```
and if a User selected some filters, for example with the key **some_filter_key1** we needed mark it somehow and don't allow user to select that filter again (it should be removed from dropdown) and to make it we should call method `this.context.filterBarStore.getDisabled()` and pass it as `disabledFilters` property into component. 

The new logic works in completely different way, we still have original registered hash of filters (as I described above) but data and values we have in the same **FilterBarStore** but in `activeFilters` array. In this case I use `groupKey` and `inputKey` properties to manage values data. It's a double array, which has `groupKey` as ID of the first array and `inputKey` as ID for the second array.

Reply:
Thanks for the explanation. I definitely like the change, but you're right that it's a big one.